extension/encoding/awslogsencodingextension/internal/unmarshaler/s3-access-log/unmarshaler.go

```diff
@@ -0,0 +1,244 @@
+			return plog.Logs{}, err
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return plog.Logs{}, fmt.Errorf("error reading log line: %w", err)
+	}
+
+	s.setResourceAttributes(resourceAttr, resourceLogs)
+	return logs, nil
+}
+
+// createLogs with the expected fields for the scope logs
+func (s *s3AccessLogUnmarshaler) createLogs() (plog.Logs, plog.ResourceLogs, plog.ScopeLogs) {
+	logs := plog.NewLogs()
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	scopeLogs.Scope().SetName(metadata.ScopeName)
+	scopeLogs.Scope().SetVersion(s.buildInfo.Version)
+	return logs, resourceLogs, scopeLogs
+}
+
+// setResourceAttributes based on the resourceAttributes
+func (s *s3AccessLogUnmarshaler) setResourceAttributes(r *resourceAttributes, logs plog.ResourceLogs) {
+	attr := logs.Resource().Attributes()
+	attr.PutStr(semconv.AttributeCloudProvider, semconv.AttributeCloudProviderAWS)
+	if r.bucketName != "" {
+		attr.PutStr(semconv.AttributeAWSS3Bucket, r.bucketName)
+	}
+	if r.bucketOwner != "" {
+		attr.PutStr(attributeAWSS3BucketOwner, r.bucketOwner)
+	}
+}
+
+// scanField gets the next value in the log line by moving
+// one space. If the value starts with a quote, it moves
+// forward until it finds the ending quote, and considers
+// that just 1 value. Otherwise, it returns the value as
+// it is.
+func scanField(logLine string) (string, string, error) {
+	if logLine == "" {
+		return "", "", io.EOF
+	}
+
+	if logLine[0] != '"' {
+		value, remaining, _ := strings.Cut(logLine, " ")
+		return value, remaining, nil
+	}
+
+	// if there is a quote, we need to get the rest of the value
+	logLine = logLine[1:] // remove first quote
+	value, remaining, found := strings.Cut(logLine, `"`)
+	if !found {
+		return "", "", fmt.Errorf("value %q has no end quote", logLine)
+	}
+
+	// Remove space after closing quote if present
+	if len(remaining) > 0 && remaining[0] == ' ' {
+		remaining = remaining[1:]
+	}
+
+	return value, remaining, nil
+}
+
+func handleLog(resourceAttr *resourceAttributes, scopeLogs plog.ScopeLogs, log string) error {
+	record := plog.NewLogRecord()
+
+	remaining := log
+	var i int
+	var value string
+	var err error
+	for i = 0; remaining != ""; i++ {
+		if i >= len(attributeNames) {
+			return errors.New("values in log line exceed the number of available fields")
+		}
+
+		value, remaining, err = scanField(remaining)
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			return err
+		}
+
+		if value == unknownField && i != fieldIndexACLRequired {
+			// acl required field can be '-' to indicate that no ACL was required
+			continue
+		}
+
+		if i == fieldIndexTime {
+			// This is the timestamp that follows a strict format
+			// "[DD/MM/YYYY:HH:mm:ss zone]". Since zone is after a
+			// space, we cut the string again to remove the zone.
+			// Zone is always UTC, so it will always be +0000.
+			_, remaining, _ = strings.Cut(remaining, " ")
+		}
```

Comment:
This feels a bit like an assumption waiting to bite us. Could we parse the timezone, even if we expect it to always be UTC? We could update scanField to check if the first character is `[` and then look for a matching `]`. Either always do it, or only when the index is `fieldIndexTime`

Reply:
Sure, now we consider the zone and don't assume that it is UTC.